planet/src/main/webapp/js/itinerary.js

```diff
@@ -21,47 +49,133 @@ function renderStartingLocation() {
     }
 }
 
-async function renderEvents() {
-    const eventsResponse = await fetch('/update-event');
-    const events = await eventsResponse.json();
+// Add an event to the firebase realtime database
+// Todo: input validation, success/failure callbacks
+async function addEvent() {
+    const eventName = document.getElementById('add-event-name').value;
+    const eventAddress = document.getElementById('add-event-address').value;
+    const eventDuration = document.getElementById('add-event-duration').value;
+    const userId = firebase.auth().currentUser.uid; 
+    
+    // Set every event's opening hours to 8am - 5pm for now
+    const openingTime = TimeRange.getTimeInMinutes(8,0);
+    const closingTime = TimeRange.getTimeInMinutes(17,0);
+
+    const eventListRef = database.ref('users/' + userId + '/currentList');
+    // Get the order number by counting existing events
+    const snap = await eventListRef.once('value');
+    const order = snap.numChildren() + 1;
+
+    //Create a new event
+    const newEventRef = eventListRef.push();
+    newEventRef.set({
+        name: eventName,
+        address: eventAddress,
+        duration: eventDuration,
+        openingTime: openingTime,
+        closingTime: closingTime,
+        order: order,
+    });
+    closeForm();
+}
+
+// Check authentication status and render the list of events if the user has
+// sign in.
+// Todo: discuss with teammates to decide what to show if user is not signed in.
+firebase.auth().onAuthStateChanged(function(user) {
+    if (user) {
+        renderEvents('currentList');
+    } else {
+        console.log('Please sign in');
+    }
+});
 
-    events.forEach((event) => {
-        let eventElement = createEventElement (event.id, event.name, 
-            event.address, event.duration);
-        document.getElementById('events').appendChild(eventElement);
+function renderEvents(listName) {
+    const userId = firebase.auth().currentUser.uid;
```

Comment:
consider making this a global variable (you instantiate it and declare it the same thing in multiple areas)

Reply:
The functions can be called from different user sessions, right? I think it makes sense to query for uid every time we need it.

Reply:
True, I'll leave it as is and see if there's a need to simplify the code in the future